### PR TITLE
Nuking SignupViewController

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're effectively removing **SignupViewController**, which has been replaced by the new Signup Flow, and is not in use anymore.

Ref. #8109

@elibud @ScoutHarris @nheagy mind taking a peek at this one?

Thanks in advance!!

🔥 🔥 🔥 🔥 🔥 🔥 

### To test:
1. Verify the unit tests run correctly
2. Verify that `Signup with Email` works as expected
3. Verify that `Signup with Google` works as expected
4. Log into a Self Hosted. Open > Me > Log In > Signup, and verify the slider shows up correctly.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Me/MeViewController.swift [hunk 1]
```diff
@@ -444,39 +444,7 @@ class MeViewController: UITableViewController, UIViewControllerRestoration {
...
-            present(controller, animated: true, completion: nil)
-
-            if let presentationController = controller.popoverPresentationController,
-                let cell = tableView.visibleCells.last {
-                presentationController.permittedArrowDirections = .any
-                presentationController.sourceView = cell
-                presentationController.sourceRect = cell.bounds
-            }
-        }
+        WordPressAuthenticator.showLogin(from: self, animated: true, showCancel: true, restrictToWPCom: true)
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/Me/MeViewController.swift]: this one is my favorite

File: WordPress/Classes/ViewRelated/NUX/SiteCreationSiteDetailsViewController.swift [hunk 2]
```diff
@@ -2,7 +2,7 @@ import UIKit
 
 class SiteCreationSiteDetailsViewController: NUXViewController, NUXKeyboardResponder {
 
-    // MARK: - SigninKeyboardResponder Properties
+    // MARK: - NUXKeyboardResponder Properties
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationSiteDetailsViewController.swift]: Nice catch!

File: WordPress/Classes/Services/SignupService.swift [hunk 3]
```diff
@@ -3,146 +3,15 @@ import CocoaLumberjack
 import WordPressShared
```
Comment (nheagy) [inline WordPress/Classes/Services/SignupService.swift]: This file has barely any contents now. Should it be renamed? It's less `SignupService` and more `GoogleSignupService` 😜

Comment (jleandroperez) [reply WordPress/Classes/Services/SignupService.swift]: Ahhhh good point sir!! Let's rename in another PR!!.

File: WordPress/Classes/ViewRelated/Me/MeViewController.swift (hunk 1 above)
Comment (jleandroperez) [reply WordPress/Classes/ViewRelated/Me/MeViewController.swift]: Wooooooo!!!

**Comment by jleandroperez** [comment]
Thanks a looot for the review sir!!. I'll rename the service in another PR (which is already written!!) so that... i don't break all the things in the process!!.

Wooooooooo!!!!

**Pull request merged**